### ISSUE TITLE
Fedora 36 cannot shutdown and poweroff if there is no keyboard attached

### ISSUE THREAD

**Issue by smuda** [coreos/fedora-coreos-tracker#1281]
**Describe the bug**
Issuing `sudo shutdown now` (or pushing the power button) does not power off the Fedora CoreOS 36 machine, but rather reboots if there are no keyboard attached.

**Reproduction steps**
Prerequisities
- Intel NUC
- Fedora CoreOS 36 (installed with OKD 4.11)
- no keyboard attached.

Login via SSH and issue command `sudo shutdown now`.

**Expected behavior**
I expect the machine to shutdown and power off.

**Actual behavior**
The master shuts down and reboots.

**System details**
 - Bare Metal

**Additional information**
The firmware of the NUCs have been upgraded to latest version, with no changes in behavior at all.
If a keyboard is attached, it works as expected.

**Comment by dustymabe** [comment]
Anything interesting in the logs of the nodes that won't shutdown properly?

> ```
> 4.11.0-0.okd-2022-08-20-022919 (with FCOS 35.20220327.3.0): Does not shutdown
> 4.11.0-0.okd-2022-07-29-154152 (with FCOS 35.20220327.3.0): Does not shutdown
> 4.10.0-0.okd-2022-07-09-073606 (with FCOS 35.20220327.3.0): Shutdown correctly
> 4.10.0-0.okd-2022-06-24-212905 (with FCOS 35.20220327.3.0): Shutdown correctly
> 4.10.0-0.okd-2022-06-10-131327 (with FCOS 35.20220327.3.0): Shutdown correctly
> 4.10.0-0.okd-2022-05-28-062148 (with FCOS 35.20220327.3.0): Shutdown correctly
> ```

This detail indicates to me that the issue is at least triggered by the OKD stack. i.e. you are using the same version of FCOS between all of these cases and some of them work and some of them don't.

The issue may indeed be in FCOS itself, but it's obviously triggered by some change in OKD. It would be nice to know what that change was so we can further investigate and fix the issue in FCOS.

cc @vrutkovs 

Additionally Fedora 35 based Fedora CoreOS is old (we don't have any streams on Fedora 35 any longer). Maybe the problem still exists in Fedora 36 based FCOS? We'd probably need to know that before spending time on investigation here.

**Comment by vrutkovs** [comment]
4.11 nodes should be F36-based, so it looks like a Fedora 36 bug to me on this particular hardware

**Comment by dustymabe** [comment]
> 4.11 nodes should be F36-based, so it looks like a Fedora 36 bug to me on this particular hardware

so I take it the information provided by @smuda is incorrect when they name the FCOS version as `35.20220327.3.0`?

**Comment by smuda** [comment]
Ah, yes. The installation of OKD starts with FCOS 35 and I didn't realize it upgrades to 36. This is when running 4.11.0-0.okd-2022-08-20-022919:

```
[core@master1 ~]$ uname --all
Linux master1.okd4.example.com 5.18.17-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Aug 11 14:36:06 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

[core@master1 ~]$ cat /etc/os-release
NAME="Fedora Linux"
VERSION="36 (CoreOS)"
ID=fedora
VERSION_ID=36
VERSION_CODENAME=""
PLATFORM_ID="platform:f36"
PRETTY_NAME="Fedora CoreOS 36"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:36"
HOME_URL="https://getfedora.org/coreos/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora-coreos/"
SUPPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
BUG_REPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=36
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=36
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="CoreOS"
VARIANT_ID=coreos
OSTREE_VERSION='411.36.202208191044-0'
DEFAULT_HOSTNAME=localhost

[core@master1 ~]$ lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         39 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  12
  On-line CPU(s) list:   0-11
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Core(TM) i7-10710U CPU @ 1.10GHz
    CPU family:          6
    Model:               166
    Thread(s) per core:  2
    Core(s) per socket:  6
    Socket(s):           1
    Stepping:            0
    Frequency boost:     enabled
    CPU(s) scaling MHz:  207%
    CPU max MHz:         1601.0000
    CPU min MHz:         400.0000
    BogoMIPS:            3199.92
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpui
                         d aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_sin
                         gle ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust sgx bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat 
                         pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d arch_capabilities
Virtualization features: 
  Virtualization:        VT-x
Caches (sum of all):     
  L1d:                   192 KiB (6 instances)
  L1i:                   192 KiB (6 instances)
  L2:                    1.5 MiB (6 instances)
  L3:                    12 MiB (1 instance)
NUMA:                    
  NUMA node(s):          1
  NUMA node0 CPU(s):     0-11
Vulnerabilities:         
  Itlb multihit:         KVM: Mitigation: VMX disabled
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Vulnerable
  Retbleed:              Vulnerable
  Spec store bypass:     Vulnerable
  Spectre v1:            Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
  Spectre v2:            Vulnerable, IBPB: disabled, STIBP: disabled, PBRSB-eIBRS: Vulnerable
  Srbds:                 Not affected
  Tsx async abort:       Not affected
```

**Comment by smuda** [comment]
So I did a shutdown of master1 on 07:38 local time and tried to find anything in the logs:

```
[core@master1 ~]$ last -5 reboot shutdown root
reboot   system boot  5.18.13-200.fc36 Tue Aug 23 07:39   still running
reboot   system boot  5.18.13-200.fc36 Tue Aug 23 07:19 - 07:38  (00:19)
reboot   system boot  5.18.13-200.fc36 Tue Aug 23 06:24 - 07:18  (00:54)
reboot   system boot  5.16.16-200.fc35 Tue Aug 23 06:22 - 06:24  (00:02)

[core@master1 ~]$ journalctl --list-boots 
 -3 23c74c50cf894e13b51116441401cf5c Tue 2022-08-23 06:14:56 CEST—Tue 2022-08-23 06:24:16 CEST
 -2 97ef89e8b5ab4503891980c372d079ee Tue 2022-08-23 06:24:45 CEST—Tue 2022-08-23 07:18:52 CEST
 -1 d24284ca50f44aa788c4a9d506fc162e Tue 2022-08-23 07:19:24 CEST—Tue 2022-08-23 07:38:37 CEST
  0 6cac46b4fa8f41d59045d3a3c0116f58 Tue 2022-08-23 07:39:13 CEST—Tue 2022-08-23 07:42:41 CEST
```

I also extracted log from journalctl with `journalctl -t kernel` but that seems to large to paste into the issue so I uploaded it [here](https://allberg-filedrop.s3.eu-north-1.amazonaws.com/master1-kernel.log).

I also extracted the unfiltered log from journalctl and uploaded it [here](https://allberg-filedrop.s3.eu-north-1.amazonaws.com/master1-journalctl.log).

Is there any other logs I can provide you with?

**Comment by dustymabe** [comment]
There doesn't seem to be anything that stands out in those logs (briefly looking). Maybe there's more information if you watch the serial console of the machine when you attempt a shutdown?

From the logs I can tell the machine is using `Linux version 5.18.13-200.fc36.x86_64`. So that is at least one piece of information.

**Comment by smuda** [comment]
As there is no serial port on a NUC and I haven't used a serial port during the last 15 years, I messed around and after a while I found a consistency. 

I use an old wireless Logitech keyboard and without it attached, five "shutdown" in a row resulted in a reboot.

With the keyboard attached, five "shutdown" in a row all resulted in a shutdown. 

If you disconnect your keyboard, would you be able to recreate my problem?

**Comment by dustymabe** [comment]
That's not really enough information to go on. Sorry.

It sounds like if you leave the keyboard connected (or not, I can't really tell from your description) that the system reboots fine.

Maybe just leave the keyboard in whatever state allows the reboot and hope the issue gets fixed in the kernel soon.

**Comment by smuda** [comment]
Sorry for not being clear enough, let me try again. 

# Prerequisities
- Intel NUC
- Fedora CoreOS 36, installed with OKD 4.11
- *no keyboard attached*.

# Objective
Shutdown the system.

# Operation
Via ssh, issue the command "sudo shutdown now".

# Expected result
The system shuts down.

# Experienced result
The system reboots.

**Comment by dustymabe** [comment]
and if you attach a keyboard the system shuts down and powers off like you expect?

**Comment by smuda** [comment]
Yes

**Comment by smuda** [comment]
I've updated the subject of the issue and the initial comment to reflect what has been discovered since then.

**Comment by travier** [comment]
Likely a kernel bug.

**Comment by smuda** [comment]
> Likely a kernel bug.

Does that mean I should file a report at Linux or do Fedora maintainers also handle the kernel? What can I do to proceed?

**Comment by smuda** [comment]
It seems this fixed somewhere between ostree 411.36.202208191044-0 and 411.36.202210130930-0.

This means that for me, with a installation of OKD 4.11.0-0.okd-2022-10-15-073651, it now works to shutdown all my machines at the same time again.

Thank you for your help!
